File: problem/Array/twoSum.py
# 给定一个整数数组 nums 和一个目标值 target，请你在该数组中找出和为目标值的那两个整数，并返回他们的数组下标。

# 你可以假设每种输入只会对应一个答案。但是，你不能重复利用这个数组中同样的元素。

# 示例:
# 给定 nums = [2, 7, 11, 15], target = 9

# 因为 nums[0] + nums[1] = 2 + 7 = 9
# 所以返回 [0, 1]

# 来源：力扣（LeetCode）
# 链接：https://leetcode-cn.com/problems/two-sum


class Solution:
    def twoSum(self, nums, target):
        hashmap = {}
        for i, val in enumerate(nums):
            complement = target - val
            if complement in hashmap:
                return [hashmap[complement], i]
            hashmap[val] = i

    def __init__(self, nums, target):
        print(self.twoSum(nums, target))


nums = [2, 7, 11, 15]
target = 9
Solution(nums, target)

File: array/two-sum.js
// https://leetcode-cn.com/problems/two-sum
// 2022-07-11

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
 var twoSum = function(nums, target) {
   const map = new Map();
   if (!nums?.length) {
     return []
   }
   for (let i = 0; i < nums.length; i++) {
     if (map.has(target-nums[i])) {
      return [map.get(target-nums[i]),i]
     }
     map.set(nums[i],i)
   }
};